proto/api.proto

```diff
@@ -3,10 +3,87 @@ syntax = "proto3";
 package api;
 
 import "geometry.proto";
-import "features.proto";
 
 option go_package = "diagonal.works/b6/proto";
 
+message TagProto {
+    string key = 1;
+    string value = 2;
+}
+
+enum FeatureType {
+    FeatureTypePoint = 0;
+    FeatureTypePath = 1;
+    FeatureTypeArea = 2;
+    FeatureTypeRelation = 3;
+    FeatureTypeCollection = 4;
+    FeatureTypeExpression = 5;
+    FeatureTypeInvalid = 6;
```

Comment:
isnt it custom to invalid at 0 so u can keep adding types in the future w/o changing indices

Reply:
You're completely right, and I've changed this, but in my defense (just so it doesn't seem totally insane), there are a couple of pros:
- There's a C++ convention of having the invalid/end value at the end of a list, so that you can write code like `for i = FeatureTypePoint; i != FeatureTypeInvalid; i++`. We don't really use the pattern anywhere though.
- Originally we used these proto values directly, without conversion, in the memory mapped compact index. The four original type values (Point, Path, Area, Relation) needed 2 bits. Adding invalid would need an extra bit, and the compact index doesn't have any invalid values, because they're filtered out during the index build, so using up slot 0 for an invalid value would waste 1 bit per feature. We don't use the proto values directly anymore though, so it's not important.